specification/mcpp/resource-manager/Microsoft.CloudPartnerProgramMembership/preview/2023-02-09-preview/mcpp.json

```diff
@@ -0,0 +1,447 @@
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": ["CloudPartnerProgramMemberships"],
+        "operationId": "CloudPartnerProgramMemberships_CancelMcppPuchase",
+        "x-ms-examples": {
+          "deleteOperations": { "$ref": "./examples/cancelMcppPurchase.json" }
+        },
+        "description": "The operation to cancel a CloudPartnerProgramMembership offer",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Canceled the purchase"
+          },
+          "204": {
+            "description": "purchase record does not exist"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "McppPurchaseDetailsResponse":{
+      "description": "MCPP Purchase details response",
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "id for the purchase resource",
+          "readOnly": true
+        },
+        "name": {
+          "description": "purchase resource name.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type, CloudPartnerProgramMembership.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "properties": { 
+          "$ref": "#/definitions/McppPurchaseDetailsResponseProperties",
+          "description": "MCPP Purchase details response properties."
+        }
+      }
+    },
+    "McppPurchaseDetailsResponseProperties": {
+      "description": "MCPP Purchase details response properties",
+      "type": "object",
+      "properties": {
+        "orderNo": {
+          "type": "string",
+          "description": "order number for the purchase",
+          "readOnly": true
+        },
+        "orderDateTime": {
+          "type": "string",
+          "description": "date time at which the order was placed",
+          "readOnly": true
+        },
+        "status": {
+          "type": "string",
+          "description": "purchase status or the order",
+          "readOnly": true
+        },
+        "operationDetails":{
+          "type": "object",
+          "description": "operation details related to the purchase",
+          "readOnly": true
+        }
+      }
+    },
+    "CloudPartnerProgramMembershipPurchaseDetailsListResponse": {
```

Comment:
Recommend: MembershipDetailsListResponse

Reply:
ack, will make this change.